### PR TITLE
Remove provider deprecations in Apache Drill

### PR DESCRIPTION
**Pull request by kunaljubce**
Relates to #44559.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: providers/src/airflow/providers/apache/beam/CHANGELOG.rst [hunk 1]
```diff
@@ -542,8 +542,7 @@ Breaking changes
 Breaking changes
 ~~~~~~~~~~~~~~~~
 
-Integration with the ``google`` provider
-````````````````````````````````````````
+Integration with the ``google`` provider.
```
Comment (jscheffl) [inline providers/src/airflow/providers/apache/beam/CHANGELOG.rst]: This change seems to be unrelated to drill, can you revert this?

Comment (kunaljubce) [reply providers/src/airflow/providers/apache/beam/CHANGELOG.rst]: @jscheffl Reverted this, thanks!

**Review by potiuk — APPROVED** [review]

File: providers/src/airflow/providers/apache/beam/CHANGELOG.rst (hunk 1 above)
Comment (jscheffl) [reply providers/src/airflow/providers/apache/beam/CHANGELOG.rst]: Thanks!
